src/Controller/TagsController.php

```diff
@@ -246,4 +246,33 @@ public function search()
             $search
         ]);
     }
+
+    /**
+     * Display list of tags for autocompletion.
+     *
+     * @param String $filter Filters the tags list with only those that contain the
+     *                       search string.
+     */
+    public function autocomplete($search)
+    {
+        $this->helpers[] = 'Tags';
+
+        $conditions = [];
+        if (!empty($search)) {
+            $conditions = [
+                'name LIKE' => "$search%"
+            ];
+        }
+        $this->paginate = [
+            'limit' => 10,
+            'fields' => ['name', 'id', 'nbrOfSentences'],
+            'order' => ['nbrOfSentences' => 'DESC'],
+            'conditions' => $conditions
+        ];
+
+        $this->autoRender = false;
+        $this->response = $this->response->withType('json');
+        $allTags = $this->paginate();
```

Comment:
I don’t think you need to use pagination here, because you are never going further than page 1. If all you need to do is to limit the number of results to 10, you can use a simple query, like this:
```php
$query = $this->Tags->find();
$query->select(['name', 'id', 'nbrOfSentences'])
      ->limit(10);
if (!empty($search)) {
    $query->where(['name LIKE' => "$search%"]);
}
$allTags = $query->all();
```

Reply:
Thank you! I actually did not know what `paginate` did, now I know.